### PR TITLE
Fix: target cleanup

### PR DESCRIPTION
**Pull request by dragonmux**
## Detailed description

In this the third of the cleanup PRs, we address a variety of code quality and consistency issues across the targets, fix variable naming style to conform to the style requirements, and attempt to correct comment style to be more or less consistent across all the targets. This also addresses a significant volume of clang-tidy lint issues.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

File: src/target/samd.c [hunk 1]
```diff
@@ -773,23 +772,18 @@ static bool samd_set_bootprot(target *t, uint16_t value, const char **argv)
...
-	while ((target_mem_read32(t, SAMD_NVMC_INTFLAG) & SAMD_NVMC_READY) == 0) {
-		if (target_check_error(t))
-			return -1;
-	}
+	if (!samd_wait_nvm_ready(t))
+		return false;
 
 	/* Modify the low word of the user row */
-	low = (low & 0xFFFFFFF8) | ((value << 0) & 0x00000007);
+	low = (low & 0xFFFFFFF8U) | ((value << 0U) & 0x00000007U);
```
Comment (esden) [inline src/target/samd.c:788]: Hex lower case?

File: src/target/samx5x.c [hunk 2]
```diff
@@ -61,137 +63,131 @@ static bool samx5x_cmd_write32(target *t, int argc, const char **argv);
...
-#define SAMX5X_CTRLB_CMD_KEY             0xA500
-#define SAMX5X_CTRLB_CMD_ERASEPAGE       0x0000
-#define SAMX5X_CTRLB_CMD_ERASEBLOCK      0x0001
-#define SAMX5X_CTRLB_CMD_WRITEPAGE       0x0003
-#define SAMX5X_CTRLB_CMD_WRITEQUADWORD   0x0004
-#define SAMX5X_CTRLB_CMD_LOCK            0x0011
-#define SAMX5X_CTRLB_CMD_UNLOCK          0x0012
-#define SAMX5X_CTRLB_CMD_PAGEBUFFERCLEAR 0x0015
-#define SAMX5X_CTRLB_CMD_SSB             0x0016
+#define SAMX5X_CTRLB_CMD_KEY             0xA500U
```
Comment (esden) [inline src/target/samx5x.c:101]: Hex lower case?

File: src/target/samx5x.c [hunk 3]
```diff
@@ -661,39 +637,34 @@ static int samx5x_set_flashlock(target *t, uint32_t value)
...
+	tc_printf(t, "Flash locked. The target must be reset for this to take effect.\n");
 	return true;
 }
 
 static bool samx5x_cmd_unlock_flash(target *t, int argc, const char **argv)
 {
 	(void)argc;
 	(void)argv;
-	if (samx5x_set_flashlock(t, 0xFFFFFFFF)) {
+	if (samx5x_set_flashlock(t, 0xFFFFFFFFU)) {
```
Comment (esden) [inline src/target/samx5x.c:659]: Hex lower case?

File: src/target/stm32f1.c [hunk 4]
```diff
@@ -163,7 +163,7 @@ bool gd32f1_probe(target *t)
 static bool at32f40_detect(target *t, const uint16_t part_id)
 {
 	// Current driver supports only *default* memory layout (256 KB Flash / 96 KB SRAM)
-	// (*) Support for external Flash for 512KB and 1024KB parts requires specific flash code (not implement)
+	// XXX: Support for external Flash for 512KB and 1024KB parts requires specific flash code (not implement)
```
Comment (esden) [inline src/target/stm32f1.c:166]: Should it say "not implemented", or "do not implement"? It is not clear.

**Review by esden — COMMENTED** [review]
It looks good for the most part. There is one comment that needs some elaboration. The hex case stuff can also be fixed separately, I am sure I have not even caught all of them myself.

File: src/platforms/stm32/rtt_if.c [hunk 5]
```diff
@@ -48,8 +48,7 @@ inline static uint32_t recv_bytes_free()
 {
 	if (recv_tail <= recv_head)
 		return RTT_DOWN_BUF_SIZE - recv_head + recv_tail - 1U;
-	else
-		return recv_tail - recv_head - 1U;
+	return recv_tail - recv_head - 1U;
```
[the remark this answers is not in the document]
Comment (dragonmux) [reply src/platforms/stm32/rtt_if.c:51]: it's a clang-tidy lint. There are only two branches, and it (correctly) highly discourages return-else patterns. It does not change the logic

File: src/target/samd.c (hunk 1 above)
Comment (dragonmux) [reply src/target/samd.c:788]: Good catch! We'll deal with this momentarily

File: src/target/stm32f1.c (hunk 4 above)
Comment (dragonmux) [reply src/target/stm32f1.c:166]: "not implemented" we believe, we'll update the comment

File: src/platforms/stm32/rtt_if.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (esden) [reply src/platforms/stm32/rtt_if.c:51]: ohh... right :facepalm: ... you are right never mind :)

File: src/target/stm32f1.c (hunk 4 above)
Comment (esden) [reply src/target/stm32f1.c:166]: thanks! :)

**Review by esden — APPROVED** [review]
LGTM Merging!

**Pull request merged**